wasmtime-wasi/src/translate.rs

```diff
@@ -28,20 +24,20 @@ unsafe fn decode_ptr(
                     Some(sum) => sum,
                     None => {
                         println!("!!! overflow");
-                        return Err(host::__WASI_EFAULT as host::__wasi_errno_t);
+                        return Err(host::__WASI_EFAULT);
                     }
                 };
                 // Check for out of bounds.
                 if last >= (*definition).current_length {
                     println!("!!! out of bounds");
-                    return Err(host::__WASI_EFAULT as host::__wasi_errno_t);
+                    return Err(host::__WASI_EFAULT);
                 }
             }
-            // Check alignment.
-            if (ptr as usize) % align != 0 {
-                println!("!!! bad alignment: {} % {}", ptr, align);
-                return Err(host::__WASI_EINVAL as host::__wasi_errno_t);
-            }
+            // // Check alignment.
+            // if (ptr as usize) % align != 0 {
+            //     println!("!!! bad alignment: {} % {}", ptr, align);
+            //     return Err(host::__WASI_EINVAL as host::__wasi_errno_t);
+            // }
```

Comment:
I've had to temporarily remove alignment checks in order to match the signature of `decode_ptr` with `dec_ptr` in `lucet-wasi`. I'm wondering what to do about this here? Even though we don't check for alignment here, we do now check in `dec_ptr_to` in `wasi-common` ([see L31](https://github.com/CraneStation/wasi-common/blob/c6db9f3dff7c25236702f3037783ade287c062fc/src/memory.rs#L31)).

Reply:
Pointer decoding is now fully handled in `wasi-common`, hence this is redundant now.